wandb/sdk/wandb_init.py

```diff
@@ -921,6 +921,16 @@ def init(self, settings: Settings, config: _ConfigParts) -> Run:  # noqa: C901
             )
         error: wandb.Error | None = None
 
+        # In shared mode, generate a unique label if not provided.
+        # The label is used to distinguish between system metrics and console logs
+        # from different writers to the same run.
+        if settings._shared and not settings.x_label:
+            # TODO: If executed in a known distributed environment (e.g. Ray or SLURM),
+            #   use the env vars to generate a label (e.g. SLURM_JOB_ID or RANK)
+            prefix = settings.host or ""
+            label = runid.generate_id()
+            settings.x_label = f"{prefix}-{label}" if prefix else label
+
```

Comment:
This function should not modify settings. All settings should be set in `make_run_settings`.

Reply:
Ok, will fix.